faker/providers/date_time/__init__.py

```diff
@@ -1661,11 +1661,14 @@ def date_time_between_dates(
             datetime_to_timestamp(datetime_start),
             datetime_to_timestamp(datetime_end),
         )
-        if tzinfo is None:
-            pick = datetime.fromtimestamp(timestamp, tzlocal())
-            pick = pick.astimezone(tzutc()).replace(tzinfo=None)
-        else:
-            pick = datetime.fromtimestamp(timestamp, tzinfo)
+        try:
+          if tzinfo is None:
+              pick = datetime.fromtimestamp(timestamp, tzlocal())
+              pick = pick.astimezone(tzutc()).replace(tzinfo=None)
+          else:
+              pick = datetime.fromtimestamp(timestamp, tzinfo)
+        except OSError:
+          raise OSError('Argument(s) out of bounds.  On windows machines dates provided must occur in the range of 1970-01-01T00:00:01 and 2038-12-31T23:59:59')
```

Comment:
I think we should prevent out-of-bounds arguments from ever happening. Can we have the check happen _before_ we even try to instantiate `timestamp`? Just check if `datetime_start` or `datetime_end` are within the bounds.

Reply:
(in advance forgive me if I'm mistaken, still a student). While that would improve performance on Windows machine wouldn't the check process just be unnecessary overhead on all other systems that can have values outside that range?

Reply:
You would check the boundaries only on windows. To do that, you can use `sys.platform`:

```python
import sys

if sys.platform == 'win32':
    # if datetime_start ....
    # if datetime_end ....
```

Besides, I'm not really worried about the impact of `if` statements. I look out mainly for for-loops, especially when nested.

Reply:
@fcurella I feel like that would be a really bad idea to only limit between 1970 and now. This issue was already resolved in another PR as well.

See https://github.com/joke2k/faker/blob/cb333b3b2950860ee6f3e57cb604f107ddda6fd3/faker/providers/date_time/__init__.py#L1543

Can't use fromtimestamp, instead need to use a timedelta. for times before 1970, it would be a negative number.

Reply:
@BrodyUnderwood1 have you verified that the issue still exists?

Reply:
It did yesterday. I'm out getting lunch now, I'll test again when I'm back.

Reply:
Change of plans probably can't today.  Maybe tomorrow